File: go-pr/48/go/code-quality-naming/file_operations.go
package main

import (
	"fmt"
	"io/ioutil"
)

type FileOperations struct{}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (fo *FileOperations) rd(f string) (string, bool) {
	c, err := ioutil.ReadFile(f)
	if err != nil {
		return "", false
	}
	return string(c), true
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (fo *FileOperations) ReadFileContent(filename string) (string, bool) {
	content, err := ioutil.ReadFile(filename)
	if err != nil {
		return "", false
	}
	return string(content), true
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (fo *FileOperations) wr(f string, d string) bool {
	err := ioutil.WriteFile(f, []byte(d), 0644)
	return err == nil
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (fo *FileOperations) WriteDataToFile(filename string, data string) bool {
	err := ioutil.WriteFile(filename, []byte(data), 0644)
	return err == nil
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (fo *FileOperations) ex(p string) bool {
	_, err := ioutil.ReadFile(p)
	return err == nil
}
// {/fact}

func main() {
	ops := &FileOperations{}
	content, success := ops.rd("test.txt")
	if success {
		fmt.Println(content)
	}
}

File: cpp-pr/46/cpp/code-quality-naming/file_operations.cpp
#include <iostream>
#include <fstream>
#include <string>

class FileOperations {
public:
    // {fact rule=code-quality-naming@v1.0 defects=1}
    bool rd(const std::string& f, std::string& c) {
        std::ifstream file(f);
        if (!file.is_open()) return false;
        
        std::string l;
        while (std::getline(file, l)) {
            c += l + "\n";
        }
        return true;
    }
    // {/fact}
    
    // {fact rule=code-quality-naming@v1.0 defects=0}
    bool readFileContent(const std::string& filename, std::string& content) {
        std::ifstream inputFile(filename);
        if (!inputFile.is_open()) return false;
        
        std::string currentLine;
        while (std::getline(inputFile, currentLine)) {
            content += currentLine + "\n";
        }
        return true;
    }
    // {/fact}
    
    // {fact rule=code-quality-naming@v1.0 defects=1}
    void wr(const std::string& f, const std::string& d) {
        std::ofstream file(f);
        file << d;
        file.close();
    }
    // {/fact}
    
    // {fact rule=code-quality-naming@v1.0 defects=0}
    void writeDataToFile(const std::string& filename, const std::string& data) {
        std::ofstream outputFile(filename);
        outputFile << data;
        outputFile.close();
    }
    // {/fact}
};

int main() {
    FileOperations ops;
    std::string content;
    ops.rd("test.txt", content);
    return 0;
}